app/models/relationship.rb

```diff
@@ -326,6 +327,19 @@ def move_to_same_project_as_main
     end
   end
 
+  def move_explainers_to_source
+    # Destroy common Explainer from target item (use destroy to log this event)
+    data = ExplainerItem.select('explainer_id').where(project_media_id: [self.source_id, self.target_id])
+    .group('explainer_id').having("count(explainer_id) = ?", 2)
+    ExplainerItem.where(explainer_id: data.map(&:explainer_id), project_media_id: self.target_id).destroy_all
+    # Move the Explainer from target to source by using update_all(as no callbacks) and then update logs
+    ExplainerItem.where(project_media_id: self.target_id).update_all(project_media_id: self.source_id)
+    # Update logs (to make item history consistent with Explainers attached to item)
+    Version.from_partition(self.source.team_id)
+    .where(event_type: 'create_explaineritem', associated_type: 'ProjectMedia', associated_id: self.target_id)
+    .update_all(associated_id: self.source_id)
+  end
```

Comment:
I'm not sure whether this would help performance wise, or whether it is just a readability enhancement, but what do you think about this?

```ruby
def move_explainers_to_source
  ExplainerItem.transaction do
    # Destroy common Explainer from target item
    explainer_ids = ExplainerItem.where(project_media_id: [self.source_id, self.target_id])
                                 .group('explainer_id')
                                 .having('COUNT(explainer_id) = ?', 2)
                                 .pluck(:explainer_id)
    ExplainerItem.where(explainer_id: explainer_ids, project_media_id: self.target_id).destroy_all

    # Move Explainers from target to source
    ExplainerItem.where(project_media_id: self.target_id).update_all(project_media_id: self.source_id)

    # Update logs for consistent history
    Version.from_partition(self.source.team_id)
           .where(event_type: 'create_explaineritem', associated_type: 'ProjectMedia', associated_id: self.target_id)
           .update_all(associated_id: self.source_id)
  end
end
```

 - Reduced query duplication by combining the selection and destruction of common Explainer items into a single query.
 - Batching operations by ensuring we're using transactions to ensure atomicity for multiple database changes.
 - Avoiding repeated .map(&:explainer_id) to avoid repeated object instantiation.

Reply:
Thanks @jayjay-w, I applied this comment